packages/hardhat-core/sample-projects/advanced-ts/.eslintrc.js

```diff
@@ -0,0 +1,41 @@
+module.exports = {
+  env: {
+    browser: false,
+    es2021: true,
+    mocha: true,
+    node: true,
+  },
+  plugins: ["@typescript-eslint"],
+  extends: [
+    "standard",
+    "plugin:prettier/recommended",
+    "plugin:node/recommended",
+  ],
+  parser: "@typescript-eslint/parser",
+  parserOptions: {
+    ecmaVersion: 12,
+  },
+  overrides: [
+    {
+      files: ["scripts/**"],
+      rules: { "no-process-exit": "off" },
+    },
+    {
+      files: ["hardhat.config.ts"],
+      globals: { hre: true },
+    },
+    {
+      files: ["hardhat.config.ts", "scripts/**", "test/**"],
+      rules: {
+        "node/no-unpublished-import": "off",
+        "node/no-unpublished-require": "off",
+      },
+    },
```

Comment:
Replace this with an `.npmignore` file that has:

```
hardhat.config.ts
test
```

The same can be done for the js version of the sample project

Reply:
Addressed in https://github.com/nomiclabs/hardhat/pull/1770/commits/e38f67b498e196ee73cb0a78a488761754223ffe